compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/LeakDetector.kt

```diff
@@ -26,22 +26,17 @@ class LeakDetector {
         weakReferences.add(WeakReference(obj))
     }
 
-    fun noLeak(): Boolean {
+    fun isAnyWasGC(): Boolean {
         System.gc()
         System.runFinalization()
-        for (weakReference in weakReferences) {
-            var count = 0
-            while (weakReference.get() != null && count < 5) {
-                System.gc()
-                System.runFinalization()
-                Thread.sleep(1000)
-                count++
-            }
-
-            if (weakReference.get() != null) {
-                return false
-            }
+        var count = 0
+        while (!weakReferences.any { it.get() == null } && count < 100) {
```

Comment:
I think
```
weakReferences.all { it.get() != null }
```
is clearer, but see previous comment) - I think it should actually be 
```
weakReferences.any { it.get() != null }
```

Reply:
It is better to use the same expression as in the `return` expression. But to make it more readable, I extracted it to a function.